client/src/index.jsx

```diff
@@ -28,30 +28,69 @@ class App extends React.Component {
     this.getUserFeed(userId);
   }
 
-  getUserFeed(userId) {
-    axios(`/feed/user/${userId}`)
+  refreshUserData(userId) {
+    this.getBalance(userId);
+    this.getGlobalFeed(this.state.globalFeed.newestTransactionId || null);
+    this.getUserFeed(userId, this.state.userFeed.newestTransactionId || null);
+  }
+
+  getUserFeed(userId, sinceId = null) {
+    let additionalData = {params: {sinceId: sinceId}}
+
+    axios(`/feed/user/${userId}`, additionalData)
       .then((response) => {
-        this.setState({
-          userFeed: response.data
-        });
+        this.prependNewTransactions('mine', response.data);
       })
       .catch((err) => {
-        console.log(err);
+        console.error(err);
       });
   }
 
-  getGlobalFeed() {
-    axios('/feed/global')
+  getGlobalFeed(sinceId = null) {
+    let additionalData = {params: {sinceId: sinceId}}
+
```

Comment:
Is it necessary to put the user in the url of the get request? This could be done as a POST to send all data including the additional data.

Reply:
Discussed in person.